src/main/java/org/springframework/data/elasticsearch/core/ReactiveElasticsearchTemplate.java

```diff
@@ -135,6 +137,57 @@ public ReactiveElasticsearchTemplate(ReactiveElasticsearchClient client, Elastic
 		return save(entity, getIndexCoordinatesFor(entity.getClass()));
 	}
 
+	@Override
+	public <T> Flux<T> saveAll(Mono<? extends Collection<? extends T>> entities, IndexCoordinates index) {
+
+		Assert.notNull(entities, "Entities must not be null!");
+
+		return entities.flatMapMany(entityList -> {
+
+			List<AdaptibleEntity<? extends T>> adaptibleEntities = entityList.stream() //
+					.map(e -> operations.forEntity(e, converter.getConversionService())) //
+					.collect(Collectors.toList());
+			Iterator<AdaptibleEntity<? extends T>> iterator = adaptibleEntities.iterator();
+			List<IndexQuery> indexRequests = adaptibleEntities.stream() //
+					.map(e -> getIndexQuery(e.getBean(), e)) //
+					.collect(Collectors.toList());
```

Comment:
why create an intermediate collection that is streamed again?

```
            List<IndexQuery> indexRequests = entityList.stream() //
                .map(e -> operations.forEntity(e, converter.getConversionService())) //
                .map(e -> getIndexQuery(e.getBean(), e)) //
                .collect(Collectors.toList());
```

Reply:
`adaptibleEntities` collection is required to create an iterator, which provides `AdaptibleEntity` instances for response